// mathparser.org-mxparser/out/mariuszgromada/math/mxparser/parsertokens/Function2Arg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Function2Arg = void 0;
/* Generated from Java with JSweet 3.0.0 - http://www.jsweet.org */
const mXparserConstants_1 = require("../mXparserConstants");
/**
 * Binary functions (2 arguments) - mXparserConstants tokens definition.
 *
 * @author         <b>Mariusz Gromada</b><br>
 * <a href="mailto:mariuszgromada.org@gmail.com">mariuszgromada.org@gmail.com</a><br>
 * <a href="http://github.com/mariuszgromada/MathParser.org-mXparser" target="_blank">mXparser on GitHub</a><br>
 *
 * @version        4.2.0
 * @class
 */
class Function2Arg {
    static INC_GAMMA_LOWER_SYN_$LI$() { if (Function2Arg.INC_GAMMA_LOWER_SYN == null) {
        Function2Arg.INC_GAMMA_LOWER_SYN = Function2Arg.INC_GAMMA_LOWER_STR + "(s,x)";
    } return Function2Arg.INC_GAMMA_LOWER_SYN; }
    static INC_GAMMA_UPPER_SYN_$LI$() { if (Function2Arg.INC_GAMMA_UPPER_SYN == null) {
        Function2Arg.INC_GAMMA_UPPER_SYN = Function2Arg.INC_GAMMA_UPPER_STR + "(s,x)";
    } return Function2Arg.INC_GAMMA_UPPER_SYN; }
    static REG_GAMMA_LOWER_SYN_$LI$() { if (Function2Arg.REG_GAMMA_LOWER_SYN == null) {
        Function2Arg.REG_GAMMA_LOWER_SYN = Function2Arg.REG_GAMMA_LOWER_STR + "(s,x)";
    } return Function2Arg.REG_GAMMA_LOWER_SYN; }
    static REG_GAMMA_UPPER_SYN_$LI$() { if (Function2Arg.REG_GAMMA_UPPER_SYN == null) {
        Function2Arg.REG_GAMMA_UPPER_SYN = Function2Arg.REG_GAMMA_UPPER_STR + "(s,x)";
    } return Function2Arg.REG_GAMMA_UPPER_SYN; }
    static REG_GAMMA_LOWER_P_SYN_$LI$() { if (Function2Arg.REG_GAMMA_LOWER_P_SYN == null) {
        Function2Arg.REG_GAMMA_LOWER_P_SYN = Function2Arg.REG_GAMMA_LOWER_P_STR + "(s,x)";
    } return Function2Arg.REG_GAMMA_LOWER_P_SYN; }
    static REG_GAMMA_UPPER_Q_SYN_$LI$() { if (Function2Arg.REG_GAMMA_UPPER_Q_SYN == null) {
        Function2Arg.REG_GAMMA_UPPER_Q_SYN = Function2Arg.REG_GAMMA_UPPER_Q_STR + "(s,x)";
    } return Function2Arg.REG_GAMMA_UPPER_Q_SYN; }
    static BINOM_COEFF_NCK_SYN_$LI$() { if (Function2Arg.BINOM_COEFF_NCK_SYN == null) {
        Function2Arg.BINOM_COEFF_NCK_SYN = Function2Arg.BINOM_COEFF_NCK_STR + "(n,k)";
    } return Function2Arg.BINOM_COEFF_NCK_SYN; }
    static PERMUTATIONS_SYN_$LI$() { if (Function2Arg.PERMUTATIONS_SYN == null) {
        Function2Arg.PERMUTATIONS_SYN = Function2Arg.PERMUTATIONS_STR + "(n,k)";
    } return Function2Arg.PERMUTATIONS_SYN; }
    static BETA_SYN_$LI$() { if (Function2Arg.BETA_SYN == null) {
        Function2Arg.BETA_SYN = Function2Arg.BETA_STR + "(x,y)";
    } return Function2Arg.BETA_SYN; }
    static LOG_BETA_SYN_$LI$() { if (Function2Arg.LOG_BETA_SYN == null) {
        Function2Arg.LOG_BETA_SYN = Function2Arg.LOG_BETA_STR + "(x,y)";
    } return Function2Arg.LOG_BETA_SYN; }
    static LOG_SINCE_$LI$() { if (Function2Arg.LOG_SINCE == null) {
        Function2Arg.LOG_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.LOG_SINCE; }
    static MOD_SINCE_$LI$() { if (Function2Arg.MOD_SINCE == null) {
        Function2Arg.MOD_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.MOD_SINCE; }
    static BINOM_COEFF_SINCE_$LI$() { if (Function2Arg.BINOM_COEFF_SINCE == null) {
        Function2Arg.BINOM_COEFF_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.BINOM_COEFF_SINCE; }
    static BINOM_COEFF_NCK_SINCE_$LI$() { if (Function2Arg.BINOM_COEFF_NCK_SINCE == null) {
        Function2Arg.BINOM_COEFF_NCK_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.BINOM_COEFF_NCK_SINCE; }
    static BERNOULLI_NUMBER_SINCE_$LI$() { if (Function2Arg.BERNOULLI_NUMBER_SINCE == null) {
        Function2Arg.BERNOULLI_NUMBER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.BERNOULLI_NUMBER_SINCE; }
    static STIRLING1_NUMBER_SINCE_$LI$() { if (Function2Arg.STIRLING1_NUMBER_SINCE == null) {
        Function2Arg.STIRLING1_NUMBER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.STIRLING1_NUMBER_SINCE; }
    static STIRLING2_NUMBER_SINCE_$LI$() { if (Function2Arg.STIRLING2_NUMBER_SINCE == null) {
        Function2Arg.STIRLING2_NUMBER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.STIRLING2_NUMBER_SINCE; }
    static WORPITZKY_NUMBER_SINCE_$LI$() { if (Function2Arg.WORPITZKY_NUMBER_SINCE == null) {
        Function2Arg.WORPITZKY_NUMBER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.WORPITZKY_NUMBER_SINCE; }
    static EULER_NUMBER_SINCE_$LI$() { if (Function2Arg.EULER_NUMBER_SINCE == null) {
        Function2Arg.EULER_NUMBER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.EULER_NUMBER_SINCE; }
    static KRONECKER_DELTA_SINCE_$LI$() { if (Function2Arg.KRONECKER_DELTA_SINCE == null) {
        Function2Arg.KRONECKER_DELTA_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.KRONECKER_DELTA_SINCE; }
    static EULER_POLYNOMIAL_SINCE_$LI$() { if (Function2Arg.EULER_POLYNOMIAL_SINCE == null) {
        Function2Arg.EULER_POLYNOMIAL_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.EULER_POLYNOMIAL_SINCE; }
    static HARMONIC_NUMBER_SINCE_$LI$() { if (Function2Arg.HARMONIC_NUMBER_SINCE == null) {
        Function2Arg.HARMONIC_NUMBER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv10;
    } return Function2Arg.HARMONIC_NUMBER_SINCE; }
    static RND_UNIFORM_CONT_SINCE_$LI$() { if (Function2Arg.RND_UNIFORM_CONT_SINCE == null) {
        Function2Arg.RND_UNIFORM_CONT_SINCE = mXparserConstants_1.mXparserConstants.NAMEv30;
    } return Function2Arg.RND_UNIFORM_CONT_SINCE; }
    static RND_UNIFORM_DISCR_SINCE_$LI$() { if (Function2Arg.RND_UNIFORM_DISCR_SINCE == null) {
        Function2Arg.RND_UNIFORM_DISCR_SINCE = mXparserConstants_1.mXparserConstants.NAMEv30;
    } return Function2Arg.RND_UNIFORM_DISCR_SINCE; }
    static ROUND_SINCE_$LI$() { if (Function2Arg.ROUND_SINCE == null) {
        Function2Arg.ROUND_SINCE = mXparserConstants_1.mXparserConstants.NAMEv30;
    } return Function2Arg.ROUND_SINCE; }
    static RND_NORMAL_SINCE_$LI$() { if (Function2Arg.RND_NORMAL_SINCE == null) {
        Function2Arg.RND_NORMAL_SINCE = mXparserConstants_1.mXparserConstants.NAMEv30;
    } return Function2Arg.RND_NORMAL_SINCE; }
    static NDIG_SINCE_$LI$() { if (Function2Arg.NDIG_SINCE == null) {
        Function2Arg.NDIG_SINCE = mXparserConstants_1.mXparserConstants.NAMEv41;
    } return Function2Arg.NDIG_SINCE; }
    static DIGIT10_SINCE_$LI$() { if (Function2Arg.DIGIT10_SINCE == null) {
        Function2Arg.DIGIT10_SINCE = mXparserConstants_1.mXparserConstants.NAMEv41;
    } return Function2Arg.DIGIT10_SINCE; }
    static FACTVAL_SINCE_$LI$() { if (Function2Arg.FACTVAL_SINCE == null) {
        Function2Arg.FACTVAL_SINCE = mXparserConstants_1.mXparserConstants.NAMEv41;
    } return Function2Arg.FACTVAL_SINCE; }
    static FACTEXP_SINCE_$LI$() { if (Function2Arg.FACTEXP_SINCE == null) {
        Function2Arg.FACTEXP_SINCE = mXparserConstants_1.mXparserConstants.NAMEv41;
    } return Function2Arg.FACTEXP_SINCE; }
    static ROOT_SINCE_$LI$() { if (Function2Arg.ROOT_SINCE == null) {
        Function2Arg.ROOT_SINCE = mXparserConstants_1.mXparserConstants.NAMEv41;
    } return Function2Arg.ROOT_SINCE; }
    static INC_GAMMA_LOWER_SINCE_$LI$() { if (Function2Arg.INC_GAMMA_LOWER_SINCE == null) {
        Function2Arg.INC_GAMMA_LOWER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.INC_GAMMA_LOWER_SINCE; }
    static INC_GAMMA_UPPER_SINCE_$LI$() { if (Function2Arg.INC_GAMMA_UPPER_SINCE == null) {
        Function2Arg.INC_GAMMA_UPPER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.INC_GAMMA_UPPER_SINCE; }
    static REG_GAMMA_LOWER_SINCE_$LI$() { if (Function2Arg.REG_GAMMA_LOWER_SINCE == null) {
        Function2Arg.REG_GAMMA_LOWER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.REG_GAMMA_LOWER_SINCE; }
    static REG_GAMMA_UPPER_SINCE_$LI$() { if (Function2Arg.REG_GAMMA_UPPER_SINCE == null) {
        Function2Arg.REG_GAMMA_UPPER_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.REG_GAMMA_UPPER_SINCE; }
    static PERMUTATIONS_SINCE_$LI$() { if (Function2Arg.PERMUTATIONS_SINCE == null) {
        Function2Arg.PERMUTATIONS_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.PERMUTATIONS_SINCE; }
    static BETA_SINCE_$LI$() { if (Function2Arg.BETA_SINCE == null) {
        Function2Arg.BETA_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.BETA_SINCE; }
    static LOG_BETA_SINCE_$LI$() { if (Function2Arg.LOG_BETA_SINCE == null) {
        Function2Arg.LOG_BETA_SINCE = mXparserConstants_1.mXparserConstants.NAMEv42;
    } return Function2Arg.LOG_BETA_SINCE; }
}
exports.Function2Arg = Function2Arg;
Function2Arg.TYPE_ID = 5;
Function2Arg.TYPE_DESC = "Binary Function";
Function2Arg.LOG_ID = 1;
Function2Arg.MOD_ID = 2;
Function2Arg.BINOM_COEFF_ID = 3;
Function2Arg.BERNOULLI_NUMBER_ID = 4;
Function2Arg.STIRLING1_NUMBER_ID = 5;
Function2Arg.STIRLING2_NUMBER_ID = 6;
Function2Arg.WORPITZKY_NUMBER_ID = 7;
Function2Arg.EULER_NUMBER_ID = 8;
Function2Arg.KRONECKER_DELTA_ID = 9;
Function2Arg.EULER_POLYNOMIAL_ID = 10;
Function2Arg.HARMONIC_NUMBER_ID = 11;
Function2Arg.RND_UNIFORM_CONT_ID = 12;
Function2Arg.RND_UNIFORM_DISCR_ID = 13;
Function2Arg.ROUND_ID = 14;
Function2Arg.RND_NORMAL_ID = 15;
Function2Arg.NDIG_ID = 16;
Function2Arg.DIGIT10_ID = 17;
Function2Arg.FACTVAL_ID = 18;
Function2Arg.FACTEXP_ID = 19;
Function2Arg.ROOT_ID = 20;
Function2Arg.INC_GAMMA_LOWER_ID = 21;
Function2Arg.INC_GAMMA_UPPER_ID = 22;
Function2Arg.REG_GAMMA_LOWER_ID = 23;
Function2Arg.REG_GAMMA_UPPER_ID = 24;
Function2Arg.PERMUTATIONS_ID = 25;
Function2Arg.BETA_ID = 26;
Function2Arg.LOG_BETA_ID = 27;
Function2Arg.LOG_STR = "log";
Function2Arg.MOD_STR = "mod";
Function2Arg.BINOM_COEFF_STR = "C";
Function2Arg.BINOM_COEFF_NCK_STR = "nCk";
Function2Arg.BERNOULLI_NUMBER_STR = "Bern";
Function2Arg.STIRLING1_NUMBER_STR = "Stirl1";
Function2Arg.STIRLING2_NUMBER_STR = "Stirl2";
Function2Arg.WORPITZKY_NUMBER_STR = "Worp";
Function2Arg.EULER_NUMBER_STR = "Euler";
Function2Arg.KRONECKER_DELTA_STR = "KDelta";
Function2Arg.EULER_POLYNOMIAL_STR = "EulerPol";
Function2Arg.HARMONIC_NUMBER_STR = "Harm";
Function2Arg.RND_UNIFORM_CONT_STR = "rUni";
Function2Arg.RND_UNIFORM_DISCR_STR = "rUnid";
Function2Arg.ROUND_STR = "round";
Function2Arg.RND_NORMAL_STR = "rNor";
Function2Arg.NDIG_STR = "ndig";
Function2Arg.DIGIT10_STR = "dig10";
Function2Arg.FACTVAL_STR = "factval";
Function2Arg.FACTEXP_STR = "factexp";
Function2Arg.ROOT_STR = "root";
Function2Arg.INC_GAMMA_LOWER_STR = "GammaL";
Function2Arg.INC_GAMMA_UPPER_STR = "GammaU";
Function2Arg.REG_GAMMA_LOWER_STR = "GammaRegL";
Function2Arg.REG_GAMMA_UPPER_STR = "GammaRegU";
Function2Arg.REG_GAMMA_LOWER_P_STR = "GammaP";
Function2Arg.REG_GAMMA_UPPER_Q_STR = "GammaQ";
Function2Arg.PERMUTATIONS_STR = "nPk";
Function2Arg.BETA_STR = "Beta";
Function2Arg.LOG_BETA_STR = "logBeta";
Function2Arg.LOG_SYN = "log(a, b)";
Function2Arg.MOD_SYN = "mod(a, b)";
Function2Arg.BINOM_COEFF_SYN = "C(n, k)";
Function2Arg.BERNOULLI_NUMBER_SYN = "Bern(m, n)";
Function2Arg.STIRLING1_NUMBER_SYN = "Stirl1(n, k)";
Function2Arg.STIRLING2_NUMBER_SYN = "Stirl2(n, k)";
Function2Arg.WORPITZKY_NUMBER_SYN = "Worp(n, k)";
Function2Arg.EULER_NUMBER_SYN = "Euler(n, k)";
Function2Arg.KRONECKER_DELTA_SYN = "KDelta(i, j)";
Function2Arg.EULER_POLYNOMIAL_SYN = "EulerPol";
Function2Arg.HARMONIC_NUMBER_SYN = "Harm(x, n)";
Function2Arg.RND_UNIFORM_CONT_SYN = "rUni(a, b)";
Function2Arg.RND_UNIFORM_DISCR_SYN = "rUnid(a, b)";
Function2Arg.ROUND_SYN = "round(x, n)";
Function2Arg.RND_NORMAL_SYN = "rNor(mean, stdv)";
Function2Arg.NDIG_SYN = "ndig(number, base)";
Function2Arg.DIGIT10_SYN = "dig10(num, pos)";
Function2Arg.FACTVAL_SYN = "factval(number, factorid)";
Function2Arg.FACTEXP_SYN = "factexp(number, factorid)";
Function2Arg.ROOT_SYN = "root(rootorder, number)";
Function2Arg.LOG_DESC = "Logarithm function";
Function2Arg.MOD_DESC = "Modulo function";
Function2Arg.BINOM_COEFF_DESC = "Binomial coefficient function, number of k-combinations that can be drawn from n-elements set";
Function2Arg.BERNOULLI_NUMBER_DESC = "Bernoulli numbers";
Function2Arg.STIRLING1_NUMBER_DESC = "Stirling numbers of the first kind";
Function2Arg.STIRLING2_NUMBER_DESC = "Stirling numbers of the second kind";
Function2Arg.WORPITZKY_NUMBER_DESC = "Worpitzky number";
Function2Arg.EULER_NUMBER_DESC = "Euler number";
Function2Arg.KRONECKER_DELTA_DESC = "Kronecker delta";
Function2Arg.EULER_POLYNOMIAL_DESC = "EulerPol";
Function2Arg.HARMONIC_NUMBER_DESC = "Harmonic number";
Function2Arg.RND_UNIFORM_CONT_DESC = "Random variable - Uniform continuous distribution U(a,b), usage example: 2*rUni(2,10)";
Function2Arg.RND_UNIFORM_DISCR_DESC = "Random variable - Uniform discrete distribution U{a,b}, usage example: 2*rUnid(2,100)";
Function2Arg.ROUND_DESC = "Half-up rounding, usage examples: round(2.2, 0) = 2, round(2.6, 0) = 3, round(2.66,1) = 2.7";
Function2Arg.RND_NORMAL_DESC = "Random variable - Normal distribution N(m,s) m - mean, s - stddev, usage example: 3*rNor(0,1)";
Function2Arg.NDIG_DESC = "Number of digits representing the number in numeral system with given base";
Function2Arg.DIGIT10_DESC = "Digit at position 1 ... n (left -> right) or 0 ... -(n-1) (right -> left) - base 10 numeral system";
Function2Arg.FACTVAL_DESC = "Prime decomposition - factor value at position between 1 ... nfact(n) - ascending order by factor value";
Function2Arg.FACTEXP_DESC = "Prime decomposition - factor exponent / multiplicity at position between 1 ... nfact(n) - ascending order by factor value";
Function2Arg.ROOT_DESC = "N-th order root of a number";
Function2Arg.INC_GAMMA_LOWER_DESC = "Lower incomplete gamma special function, \u03b3(s,x)";
Function2Arg.INC_GAMMA_UPPER_DESC = "Upper incomplete Gamma special function, \u0393(s,x)";
Function2Arg.REG_GAMMA_LOWER_DESC = "Lower regularized P gamma special function, P(s,x)";
Function2Arg.REG_GAMMA_UPPER_DESC = "Upper regularized Q Gamma special function, Q(s,x)";
Function2Arg.PERMUTATIONS_DESC = "Number of k-permutations that can be drawn from n-elements set";
Function2Arg.BETA_DESC = "The Beta special function B(x,y), also called the Euler integral of the first kind";
Function2Arg.LOG_BETA_DESC = "The Log Beta special function ln B(x,y), also called the Log Euler integral of the first kind, ln B(x,y)";
Function2Arg["__class"] = "org.mariuszgromada.math.mxparser.parsertokens.Function2Arg";
